libtsuba/src/RDGPartHeader.cpp

```diff
@@ -74,7 +74,7 @@ namespace tsuba {
 
 // Regex for partition files
 const std::regex kPartitionFile(
-    "part_(?:(vers[0-9A-Za-z_]+))_(?:(rdg[0-9A-Za-z-]*))_(?:(node[0-9]*))$");
+    "part_vers(?:([0-9A-Za-z_]+))_(?:(rdg[0-9A-Za-z-]*))_node(?:([0-9]*))$");
```

Comment:
this regex is still weird to me why do we:
 * Allow arbitrary characters in the version but not in the node number?
 * why are all these non match groups used?
 * Why is the node id allowed to be empty?
 I think a clearer regex would be:
```
"part_vers([0-9]+)_(rdg[0-9A-Za-z-]*)_node([0-9]+)$"
```

Reply:
I'll be honest, I'm pretty bad with regex. I kind of went with something that got the job done to some extent. I think your restrictions on the regex make sense. I'll implement this.